src/Compilers/CSharp/Portable/Symbols/Metadata/PE/TupleTypeDecoder.cs

```diff
@@ -230,9 +252,9 @@ private ImmutableArray<TypeSymbolWithAnnotations> DecodeTypeArguments(ImmutableA
             for (int i = typeArgs.Length - 1; i >= 0; i--)
             {
                 var typeArg = typeArgs[i];
-                var decoded = DecodeType(typeArg.TypeSymbol);
+                var decoded = DecodeTypeInternal(typeArg);
                 anyDecoded |= !ReferenceEquals(decoded, typeArg.TypeSymbol);
```

Comment:
>!ReferenceEquals(decoded, typeArg.TypeSymbol); [](start = 30, length = 46)

It looks like this line does not do what it is supposed to do after the changes in this function.